examples/connectivity/plot_ica_neurovault.py

```diff
@@ -0,0 +1,121 @@
+"""
+NeuroVault cross-study ICA maps.
+================================
+
+This example shows how to download statistical maps from
+NeuroVault, label them with NeuroSynth terms,
+and compute ICA components across all the maps.
+
+See :func:`nilearn.datasets.fetch_neurovault` documentation for more details.
+"""
+# Author: Ben Cipollini
+# License: BSD
+# Ported from code authored by Chris Filo Gorgolewski, Gael Varoquaux
+# https://github.com/NeuroVault/neurovault_analysis
+
+import os.path as op
+import warnings
+warnings.simplefilter('error', RuntimeWarning)  # Catch numeric issues in imgs
+
+import numpy as np
+from matplotlib import pyplot as plt
+from scipy import stats
+from sklearn.decomposition import FastICA
+
+from nilearn import datasets
+from nilearn.image import new_img_like
+from nilearn.input_data import NiftiMasker
+from nilearn._utils import check_niimg
+from nilearn.plotting import plot_stat_map
+
+
+def clean_img(img):
+    """ Remove nan/inf entries."""
+    img = check_niimg(img)
+    img_data = img.get_data()
+    img_data[np.isnan(img_data)] = 0
+    img_data[np.isinf(img_data)] = 0
+    return new_img_like(img, img_data)
+
+
+def cast_img(img, dtype=np.float32):
+    """ Cast image to the specified dtype"""
+    img = check_niimg(img)
+    img_data = img.get_data().astype(dtype)
+    return new_img_like(img, img_data)
+
+
+### Get image and term data ###################################################
+# Download 100 matching images
+ss_all = datasets.fetch_neurovault(max_images=100,  # Use np.inf for all imgs.
+                                   map_types=['F map', 'T map', 'Z map'],
+                                   fetch_terms=True)
+images, collections = ss_all['images'], ss_all['collections']
+term_scores = ss_all['terms']
+
+# Clean & report term scores
+terms = np.array(term_scores.keys())
+term_matrix = np.asarray(term_scores.values())
```

Comment:
constructing arrays like this from dict.keys() and dict.values() doesn't seem to work for me in python 3, because these functions don't return lists anymore but dict_keys and dict_values objects. Maybe we should turn them into lists first, or access them in another way?